### DISCUSSION TITLE
Can I add pure comments to my code to functions in classes that don't do anything if they're not called from somewhere within the bundle

### DISCUSSION THREAD

**Discussion by danieltroger** [parcel-bundler/parcel#6007] [Q&A] [answered]
... and parcel forwards them to terser and terser nukes said functions?

I have a class which I'm going to rewrite eventually, but there are a bunch of functions in there which are only used in some old projects, and they get included every time in my bundle. Is there a way to get them stripped?

**Comment by mischnic** [comment, marked as answer]
Terser isn't smart enough to detect unused class methods (if that's what you're asking). You'd need something like Closure compiler.

**Reply by danieltroger** [reply]
Yeah that's exactly what I'm asking. Why is terser so dumb smh 😤 I like terser tho 😔

Do you recommend switching to closure compiler? Is it like terser but better or is it trying too hard to be smart and ending up being worse than a couple kilobytes extra?

Can it be used easily with parcel?

**Reply by mischnic** [reply]
It's just that determining what methods are used at runtime is more complex

I think you have to use https://developers.google.com/closure/compiler/docs/compilation_levels#advanced_optimizations to enable this, which means that you need to add more annotations for closure compiler.

So it's probably not easy

**Reply by danieltroger** [reply]
Hm yeah that beast seems hard to tame. Also it's not as easy to hack if it does something wrong because it's written in java.

I'll stay with terser and try to make my classes modular somehow. That unused stuff in modules now gets removed by parcel adding a pure comment already goes a long way, thanks!

**Reply by danieltroger** [reply]
BTW I think it doesn't work sometimes in IE11 builds - stuff that's been imported and not used is still in the bundle. At least it was some time ago. But I haven't put effort into reproducing
